### PR TITLE
Make sure only active Sales Order Candidates are processed

### PR DESCRIPTION
**Pull request by riadmen**
https://github.com/metasfresh/metasfresh/issues/17219

### CODE REVIEW

File: backend/de.metas.salescandidate.base/src/main/java/de/metas/ordercandidate/process/C_OLCandEnqueueForSalesOrderCreation.java [hunk 1]
```diff
@@ -46,11 +51,18 @@ protected String doIt() throws Exception
 		Check.assume(olCandProcessorId > 0, "olCandProcessorId > 0");
 
 		final PInstanceId userSelectionId = queryBL.createQueryBuilder(I_C_OLCand.class)
+				.addEqualsFilter(I_C_OLCand.COLUMNNAME_IsActive, true)
```
Comment (mrie04) [inline backend/de.metas.salescandidate.base/src/main/java/de/metas/ordercandidate/process/C_OLCandEnqueueForSalesOrderCreation.java:54]:
```suggestion
.addOnlyActiveRecordsFilter()
```

File: backend/de.metas.salescandidate.base/src/main/java/de/metas/ordercandidate/process/C_OLCandEnqueueForSalesOrderCreation.java [hunk 2]
```diff
@@ -46,11 +51,18 @@ protected String doIt() throws Exception
...
 				.addEqualsFilter(I_C_OLCand.COLUMNNAME_Processed, false)
 				.filter(getProcessInfo().getQueryFilterOrElseTrue())
 				.create()
 				.createSelection();
 
+		if (userSelectionId == null)
+		{
+			final ITranslatableString msg = Services.get(IMsgBL.class).getTranslatableMsgText(MSG_OL_CANDENQUEUE_FOR_SALES_ORDER_CREATION_NO_VALID_RECORD_SELECTED);
+			throw new AdempiereException(msg).markAsUserValidationError();
```
Comment (mrie04) [inline backend/de.metas.salescandidate.base/src/main/java/de/metas/ordercandidate/process/C_OLCandEnqueueForSalesOrderCreation.java:63]:
```suggestion
throw new AdempiereException(MSG_OL_CANDENQUEUE_FOR_SALES_ORDER_CREATION_NO_VALID_RECORD_SELECTED).markAsUserValidationError();
```

**Review by mrie04 — CHANGES_REQUESTED** [review]
pls check my suggestions

**Review by mrie04 — APPROVED** [review]
lgtm

**Comment by metas-ts** [comment]
@MergifyIO copy release

**Pull request merged**